File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagram = defaultdict(list)

        for word in strs:
            count = [0] * 26
            for i in word:
                count[ord(i) - ord('a')] += 1

            key = tuple(count)
            anagram[key] = anagram.get(key, []) + [word]

        return [value for value in anagram.values()]
            

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();

        for(String s : strs){
            int[] count = new int[26];
            for(char c: s.toCharArray()){
                int index = c - 'a';
                count[index] += 1;
            }
            String key = Arrays.toString(count);
            if (!map.containsKey(key)) { map.put(key,new ArrayList<>()); }
            map.get(key).add(s);
        }

        List<List<String>> res = new ArrayList<>();
        for (List<String> v : map.values()){
            res.add(v);
        }
        return res;
    }
}